### PR TITLE
Add support for os_network module

### LINKED ISSUE #818: Openstack: Networking: Add support to os_network module
Add support for openstack os_network_module ansible in linchpin topologies
ref: https://docs.ansible.com/ansible/2.6/modules/os_network_module.html#os-network-module
part of #786

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary:
- Updated examples and documentation for os_network role
- Added test for os_network
- Updated playbooks to support os_network ansible module
- Updated schema to support os_network module

fixes #818

### CODE REVIEW

**Comment by samvarankashyap** [comment]
[test]

File: linchpin/provision/roles/openstack/files/schema.json [hunk 1]
```diff
@@ -132,7 +132,30 @@
...
+                    "interface": { "type": "string", "required": false, "allowed": ["public", "internal", "admin"] },
+                    "key": { "type": "string", "required": false },
+                    "name": { "type": "string", "required": true},
+                    "project": { "type": "string", "required": false},
+                    "provider_network_type": { "type": "string", "required": false},
+                    "provider_physical_network": { "type": "string", "required": false},
+                    "provider_segmentation_id": { "type": "string", "required": false},
+                    "region_name": { "type": "string", "required": false},
+                    "shared": { "type": "boolean", "required": false},
+		    "timeout": { "type": "number", "required": false },
```
Comment (14rcole) [inline linchpin/provision/roles/openstack/files/schema.json]: nit: can you fix the indentation on these two lines?

File: linchpin/provision/roles/openstack/tasks/provision_os_network.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+   - auth_var == ""
+
+- name: "Append outputitem to topology_outputs"
+  set_fact:
+    topology_outputs_os_network: "{{ topology_outputs_os_network + [ res_def_output_unq ] }}"
+  when:  
+    - is_unique
+    - auth_var == ""
+
+- name: "provision/deprovision os_network stack"
```
Comment (p3ck) [inline linchpin/provision/roles/openstack/tasks/provision_os_network.yml]: Can we make the task names unique?  maybe add w/auth on this one?

File: docs/source/examples/workspaces/openstack/topologies/os-network.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
```
Comment (p3ck) [inline docs/source/examples/workspaces/openstack/topologies/os-network.yml]: Is there a pinfile that references this topology?

File: linchpin/provision/roles/openstack/tasks/provision_os_network.yml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    wait: yes
+  register: res_def_output_nunq
+  no_log: "{{ not debug_mode }}"
+  when: 
+    - not is_unique
+    - auth_var != ""
+
+- name: "Append outputitem to topology_outputs"
+  set_fact:
+    topology_outputs_os_network: "{{ topology_outputs_os_network + [ res_def_output_unq ] }}"
```
Comment (p3ck) [inline linchpin/provision/roles/openstack/tasks/provision_os_network.yml]: Is this supposed to be res_def_output_nunq ?

File: docs/source/examples/workspaces/openstack/topologies/os-network.yml (hunk 3 above)
Comment (samvarankashyap) [reply docs/source/examples/workspaces/openstack/topologies/os-network.yml]: No. However , This topology is used as a example in documentation.

**Comment by samvarankashyap** [comment]
@p3ck : bump

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**